### PR TITLE
[Compute] Better error handling of external tokens

### PR DESCRIPTION
**Pull request by qwordy**
**Description**

Resolve https://github.com/Azure/azure-cli/issues/16391

External tokens in Track 2 SDk have problem. I am still investigating why it is None.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -3273,7 +3273,14 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
...
         _, _, _, external_tokens = cred.get_all_tokens('https://management.azure.com/.default')
-        external_bearer_token = external_tokens[0][0] + ' ' + external_tokens[0][1]
+        if external_tokens:
+            external_token = external_tokens[0]
+            if len(external_token) >= 2:
+                external_bearer_token = external_token[0] + ' ' + external_tokens[1]
+            else:
+                logger.warning('Getting external tokens failed.')
+        else:
+            logger.warning('Getting external tokens failed.')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:3283]: is this a warning or error? It will cause command failed at later right?

**Comment by dkegel-sigsci** [comment]
This got me up and running again.

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:3283]: Finally, it will fail in service side if service requires external tokens.

**Review by yungezz — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -3273,7 +3273,14 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
         cred, _, _ = profile.get_login_credentials(resource=resource,
                                                    aux_subscriptions=aux_subscriptions)
         _, _, _, external_tokens = cred.get_all_tokens('https://management.azure.com/.default')
-        external_bearer_token = external_tokens[0][0] + ' ' + external_tokens[0][1]
+        if external_tokens:
+            external_token = external_tokens[0]
+            if len(external_token) >= 2:
+                external_bearer_token = external_token[0] + ' ' + external_tokens[1]
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:3279]:
```suggestion
                external_bearer_token = external_token[0] + ' ' + external_token[1]
```
Should it be `external_token[1]`?

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (eprigorodov) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:3283]: This `logger.warning()` goes to the stderr and breaks our Azure DevOps pipeline, which now thinks that `AzureCLI` task printed some error. The command itself (`az sig image-version create`) works fine as it did before, returns JSON result with   `"provisioningState": "Succeeded"`.

**Pull request merged**